modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultIRTreeToASMBytesPhase.java

```diff
@@ -1216,6 +1216,10 @@ public void visitConstant(ConstantNode irConstantNode, WriteScope writeScope) {
              */
             String fieldName = irConstantNode.getDecorationValue(IRDConstantFieldName.class);
             Type asmFieldType = MethodWriter.getType(irConstantNode.getDecorationValue(IRDExpressionType.class));
+            if (asmFieldType == null) {
+                throw irConstantNode.getLocation()
```

Comment:
Is it correct to say this should never happen outside of bugs?

Reply:
Correct. I just wanted to get a better error message for my debugging. Is ok? Should I throw the exception in another way?

Reply:
It's good. I just wanted to make sure I wasn't missing something user related.